src/rustdoc_gen.rs

```diff
@@ -110,6 +110,65 @@ impl<'a> CrateSource<'a> {
         all_crate_features.extend(self.regular_features());
         all_crate_features.into_iter().collect()
     }
+
+    /// Some feature names usually mean, that the feature is breaking semver by design.
+    /// This function filters them out.
+    ///
+    /// Feature names not included by this function:
+    ///  - `unstable`
+    ///  - `nightly`
+    ///  - `bench`
+    ///  - `no_std`
+    ///  - features with prefix `__`
+    fn heuristically_included_features(&self) -> Vec<String> {
+        let features_ignored_by_default = std::collections::HashSet::from([
+            String::from("unstable"),
+            String::from("nightly"),
+            String::from("bench"),
+            String::from("no_std"),
+        ]);
+
+        let determine = |feature_name: &String| {
+            !features_ignored_by_default.contains(feature_name) && !feature_name.starts_with("__")
+        };
+
+        self.all_features().into_iter().filter(determine).collect()
+    }
+
+    pub(crate) fn feature_list_from_config(
+        &self,
+        global_config: &mut GlobalConfig,
+        feature_config: &FeatureConfig,
+    ) -> Vec<String> {
+        let all_features: std::collections::HashSet<String> =
+            self.all_features().into_iter().collect();
+
+        let result = [
+            match feature_config.base_features {
+                BaseFeatures::All => self.all_features(),
+                BaseFeatures::Heuristic => self.heuristically_included_features(),
+                BaseFeatures::Default | BaseFeatures::None => vec![],
+            },
+            feature_config.extra_features.clone(),
+        ]
+        .concat();
+
+        result
+            .into_iter()
+            .filter(|feature_name| {
+                if !all_features.contains(feature_name) && feature_config.ignore_non_existing {
+                    global_config
+                        .shell_warn(format!(
+                            "Feature {feature_name} is not present in the baseline."
+                        ))
+                        .expect("print failed");
+                    false
```

Comment:
How does this code know that it's being called in a "baseline" context? This doesn't seem clear or obvious to me.

Reply:
Yup, I am renaming the field to `is_baseline`